### PR TITLE
refactor(sdk): Use parameters directly rather than via `locals()`

### PR DESCRIPTION
**Pull request by timoffex**
Description
-----------

Make the code easier to read. We definitely shouldn't be using `locals()` as much as we are!

- [x] I updated CHANGELOG.md, or it's not applicable

Testing
-------
Wait for CI.

The functions are private, so we should be able to freely change signatures. There are no additional code references I could find. It could break things if we do weird dynamic stuff somewhere instead of calling functions directly---but then it's the calling code that's at fault and should be fixed.

### CODE REVIEW

File: wandb/sdk/wandb_login.py [hunk 1]
```diff
@@ -115,22 +121,32 @@ def __init__(self):
         self._key = None
         self._relogin = None
 
-    def setup(self, kwargs):
-        self.kwargs = kwargs
+    def setup(
+        self,
+        *,
+        anonymous: Optional[Literal["must", "allow", "never"]] = None,
```
Comment (kptkin) [inline wandb/sdk/wandb_login.py:127]: https://github.com/wandb/wandb/blob/main/wandb/sdk/wandb_settings.py#L1023 interesting that here we validate "true" and "false" as well, but docs don't list these...

Comment (kptkin) [reply wandb/sdk/wandb_login.py:127]: https://docs.wandb.ai/ref/python/login

Comment (timoffex) [reply wandb/sdk/wandb_login.py:127]: hmm yeah.. I wonder if people use it. In this PR I'd like to preserve the current behavior.

**Comment by timoffex** [comment]
Looks like tests pass now --- the remaining failures are flakes.

File: wandb/sdk/wandb_login.py (hunk 1 above)
Comment (kptkin) [reply wandb/sdk/wandb_login.py:127]: agreed! also "true", "false" seems confusing... (if we ever had that i would depreciate it)

**Review by kptkin — APPROVED** [review]
you are a brave soul! ✌️

**Pull request merged**